### PR TITLE
Clean up input dir after finishing all jobs in a repo

### LINKED ISSUE #537: Docker volumes size increases significantly after importing all projects
I have noticed that the deployment to my staging server has been starting to fail frequently due to disk space.
Checking the disk usage I have found that the processor and gitea volumes are taking up ~**25GB** which causes image pull operations to fail.  

We can either:
1. upgrade the server,
2.  delete the repos from the processor in a clean-up job.

I prefer the second option.

##

```console
$ docker system df -v
Images space usage:

REPOSITORY   TAG       IMAGE ID   CREATED   SIZE      SHARED SIZE   UNIQUE SIZE   CONTAINERS

Containers space usage:

CONTAINER ID   IMAGE     COMMAND   LOCAL VOLUMES   SIZE      CREATED   STATUS    NAMES

Local Volumes space usage:

VOLUME NAME                                                        LINKS     SIZE
kitspace-abdo-dev_gitea-data                                       0         8.128GB <---------
fb020e97396eef361bf914971fca183126044a8554e38dd1858e223cd8428716   0         520.1kB
kitspace-abdo-dev_postgres-data                                    0         76.63MB
kitspace-abdo-dev_processor-data                                   0         16.84GB <---------
kitspace-abdo-dev_meilisearch-data                                 0         71.21MB
9cb3800976bc29e7db7bb37e8f98f70322d695d0fa2994f9efdf4c1cce18e328   0         21.27kB
```

### PR DESCRIPTION
**Pull request by AbdulrhmnGhanem**
Use bullmq flows to clean up the dir after finishing all jobs for the repo.

##
Fixes #537

### CODE REVIEW

File: processor/src/job.d.ts [hunk 1]
```diff
@@ -1,16 +1,22 @@
 import { Job as BullMQJob } from 'bullmq'
 import { KitspaceYaml } from './kitspaceYaml'
 
-export interface JobData {
-  giteaId: string | null
-  kitspaceYaml: KitspaceYaml
-  originalUrl: string
+export interface RepoJobsData {
```
[the remark this answers is not in the document]
Comment (AbdulrhmnGhanem) [reply processor/src/job.d.ts]: Done!

File: processor/src/queues.ts [hunk 2]
```diff
@@ -50,11 +51,49 @@ const processIBOMQueue = new bullmq.Queue('processIBOM', {
...
+        data,
+        name: 'projectAPI',
+        opts: { jobId: projectOutputDir },
+        queueName: q.name,
+      }))
+    })
+    .flat()
+
+  await repoProcessingFlow.add({
+    children: childrenJobs,
```
Comment (kasbah) [inline processor/src/queues.ts:88]: Grammatically correct would be `childJobs`.

File: processor/src/queues.ts [hunk 3]
```diff
@@ -50,11 +51,49 @@ const processIBOMQueue = new bullmq.Queue('processIBOM', {
...
+    data: jobData,
+    name: 'projectAPI',
+    // We have to use unique jobId for the cleanup job, whenever we recreate it;
+    // There are case where the repo gets re-cloned and if we use the same jobId
+    // the cleanup job will be ignore.
```
Comment (kasbah) [inline processor/src/queues.ts:93]:
```suggestion
    // the cleanup job will be ignored.
```

File: processor/src/queues.ts [hunk 4]
```diff
@@ -50,11 +51,49 @@ const processIBOMQueue = new bullmq.Queue('processIBOM', {
...
+      }))
+    })
+    .flat()
+
+  await repoProcessingFlow.add({
+    children: childrenJobs,
+    data: jobData,
+    name: 'projectAPI',
+    // We have to use unique jobId for the cleanup job, whenever we recreate it;
+    // There are case where the repo gets re-cloned and if we use the same jobId
```
Comment (kasbah) [inline processor/src/queues.ts:92]: Is re-cloned the right word here? Who's cloning what from where? What's the actual case where the `cleanUp` would get re-triggered on the same repo & commit?

File: processor/src/job.d.ts [hunk 5]
```diff
@@ -1,16 +1,22 @@
...
+  originalUrl: string
   outputDir: string
+  ownerName: string
+  repoName: string
+  repoDescription: string
+}
+
+export interface ProjectJobData
+  extends Omit<RepoJobData, 'kitspaceYamlArray' | 'repoDescription'> {
+  kitspaceYaml: KitspaceYaml
```
Comment (kasbah) [inline processor/src/job.d.ts:19]: Did prettier do this formatting? Is it being checked?

File: processor/src/queues.ts (hunk 2 above)
Comment (AbdulrhmnGhanem) [reply processor/src/queues.ts:88]: Done!

File: processor/src/job.d.ts (hunk 5 above)
Comment (AbdulrhmnGhanem) [reply processor/src/job.d.ts:19]: Yes, here is the output for `yarn fmt`

```
$ yarn prettier --write
$ prettier '*.json' src/ test/ --write
.eslintrc.json 17ms
package.json 4ms
tsconfig.json 2ms
src/app.ts 23ms
src/env.ts 14ms
src/events.ts 5ms
src/giteaDB.ts 31ms
src/job.d.ts 5ms   <---------------------------- perttier formats .d.ts files too.
src/kitspaceYaml.ts 21ms
src/log.ts 4ms
src/main.ts 5ms
src/meili.ts 4ms
src/queues.ts 27ms
src/redisConnection.ts 3ms
src/s3.ts 24ms
src/shell.ts 6ms
src/tasks/addToSearch.ts 9ms
src/tasks/cleanUp.ts 4ms
src/tasks/processBOM/get_partinfo.ts 4ms
src/tasks/processBOM/index.ts 14ms
src/tasks/processGerbers/board_builder.ts 10ms
src/tasks/processGerbers/findGerberFiles.ts 9ms
src/tasks/processGerbers/index.ts 38ms
src/tasks/processIBOM/index.ts 11ms
src/tasks/processInfo.ts 3ms
src/tasks/processKicadPCB/index.ts 13ms
src/tasks/processPCB.ts 4ms
src/tasks/processReadme/index.ts 9ms
src/tasks/processReadme/renderMarkdown.ts 7ms
src/tasks/processReadme/urlTransformer.ts 10ms
src/tasks/writeKitspaceYaml.ts 8ms
src/utils.ts 15ms
src/watcher.ts 10ms
src/workers.ts 15ms
test/integration/projects.test.ts 51ms
test/unit/readme_processing.test.ts 10ms
Done in 1.14s.
```

**Pull request merged**